docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+
+### 3.5. Storage Component
+
+![Storage Class Diagram](images/Storage.png)
+
+A Storage object is created by the KitchenHelper class to handle the loading and saving of ingredients, recipes and chores data.
+
+The Storage() method acts as a constructor with filepaths to local save files for ingredients, recipes and chores data.
+
+The getIngredientData(), getRecipeData() and getChoreData() methods are used to read saved data from local files into the current session of KitchenHelper. loadingIngredients() and loadingRecipeItems() methods are called in getIngredientData() and getRecipeData() respectively to sort out which Ingredient object class each object belongs to.
+
+The saveIngredientData(), saveRecipeData() and saveChoreData() methods write the current state of KitchenHelper into the local save files by calling them in command classes such as AddChoreCommand and DeleteIngredientCommand.
+
+### 3.6. Common Classes 
+Classes used by multiple components are in the `seedu.kitchenhelper.object` package.
+
+## 4. Implementation
+This section describes some details on how the features are being implemented. All recipe/ ingredient/ chore-related features can be broken down into 4 distinct functionality, addition, listing, deletion and searching.
+
+### 4.1.Ingredient-related Features
+#### 4.1.1. Addition of ingredient
+
+The addition of the ingredient feature allows the user to keep track of the ingredients in the ingredient’s list.   
+For example, `addingredient /n beef /c meat /q 2 /p 20 /e 18/02/2020` will add the ingredient `beef` 
+which have the following attributes:  category `meat`, quantity `2`, price `$20` and expiry `18/02/2020`  
+
+##### Implementation
+
+![addIngredientCommand](images/addIngredientCommand.png) 
+
+The following steps explained “Sequence diagram for an example `addingredient` command”:  
+1. The user enters `addingredient /n beef /c meat /q 2 /p 20 /e 18/02/2020`.  
+2. `KitchenHelper` calls `Parser#parseUserCommand()`.  
+3. `Parser#parseUserCommand()` will call its own method `Parser#prepareAddIngredient()`.  
+4. `Parser#prepareAddIngredient()` will first validate the attributes and create an object `AddIngredientCommand` with the attributes if successful.  
+5. `KitchenHelper` calls it own method `executeCommand()` to execute the method in `AddIngredientCommand#execute()`.  
+6. On `AddIngredientCommand#execute()`, ingredient is added and return of the message.  
+
+#### 4.1.2. List all/ specific ingredient(s)
+The list feature allows showing details of Ingredients added by the user.  All ingredients added will be shown in a sorted order, by expiry, and shown by categories. The function will require a valid string , which belongs to `all/dairy/drink/fruit/meat/miscellaneous/staple/vegetable`, 
+to be added as a parameter. Failure to do so will trigger an exception where the user will be notified of an invalid command and the syntax of the listing of the ingredients will be displayed. 
+
+##### Implementation
```

Comment:
The initial state diagram is very useful! As a user, I can know immediately what objects would be instantiated and what objects aren't for every command